contracts/Uniswap/UniswapAnchoredView.sol

```diff
@@ -223,18 +223,18 @@ contract UniswapAnchoredView is UniswapConfig {
         address uniswapMarket = config.uniswapMarket;
         uint cumulativePrice = currentCumulativePrice(config);
 
-        Observation storage newObservation = newObservations[uniswapMarket];
-        Observation storage oldObservation = oldObservations[uniswapMarket];
+        Observation memory newObservation = newObservations[uniswapMarket];
+        Observation memory oldObservation = oldObservations[uniswapMarket];
 
-        // Update new and old observations if elapsed time is bigger or equal to anchor period
+        // Update new and old observations if elapsed time is greater than or equal to anchor period
         uint timeElapsed = block.timestamp - newObservation.timestamp;
         if (timeElapsed >= anchorPeriod) {
             emit UniswapWindowUpdate(uniswapMarket, oldObservation.timestamp, newObservation.timestamp, oldObservation.acc, newObservation.acc);
```

Comment:
the event also feels like it should have the new information in it, not the old information. Basically if this read from storage at the end, it would be right, or if it was using memory then V
`
            emit UniswapWindowUpdate(uniswapMarket, newObservation.timestamp, block.timestamp, newObservation.acc, cumulativePrice);
`